.github/workflows/sdk_build.yml

```diff
@@ -15,61 +15,119 @@ on:
-          dotnet test --configuration release /p:CollectCoverage=true /p:CoverletOutputFormat=opencover /p:GITHUB_ACTIONS=false
-          dotnet clean # remove the artifacts so they can be rebuild with deterministic builds enabled
-      - name: Build solution - release
+      - name: Build
         run: dotnet build --configuration release
-      - name: Generate Nuget Packages - release
+      - name: Generate Packages
         run: dotnet pack --configuration release
-      - name: upload artifacts
+      - name: Upload packages
         uses: actions/upload-artifact@master
         with:
-          name: release_drop
+          name: packages
           path: ${{ env.NUPKG_OUTDIR }}
-      - name: Upload test coverage
-        uses: codecov/codecov-action@v1
-      - name: Publish binaries to github for tags
-        if: startswith(github.ref, 'refs/tags/v')
-        run: |
-          sudo npm install --silent --no-progress -g github-release-cli@1.3.1
-          
-          # Parse repository to get owner and repo names
-          OWNER_NAME="${GITHUB_REPOSITORY%%/*}"
-          REPO_NAME="${GITHUB_REPOSITORY#*/}"
-          
-          # Get the list of files
-          RELEASE_ARTIFACT=(${{ env.NUPKG_OUTDIR }}/*)
-          
-          export GITHUB_TOKEN=${{ secrets.DAPR_BOT_TOKEN }}
-          echo "Uploading Nuget packages to GitHub Release"
-          github-release upload \
-            --owner $OWNER_NAME \
-            --repo $REPO_NAME \
-            --body "Release dapr dotnet SDK v${REL_VERSION}" \
-            --tag "v${REL_VERSION}" \
-            --name "Dapr dotnet SDK v${REL_VERSION}" \
-            --prerelease true \
-            ${RELEASE_ARTIFACT[*]}
-      - name: Publish nuget packages to nuget.org
-        if: startswith(github.ref, 'refs/tags/v') && !(endsWith(github.ref, '-rc') || endsWith(github.ref, '-dev') || endsWith(github.ref, '-prerelease'))
-        run: |
-          dotnet nuget push "${{ env.NUPKG_OUTDIR }}/Dapr*.nupkg" --skip-duplicate --api-key ${{ secrets.NUGETORG_DAPR_API_KEY }} --source https://api.nuget.org/v3/index.json
+
+  test:
+    name: Test .NET ${{ matrix.dotnet-version }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        dotnet-version: ['3.1', '5.0']
+        include:
+        - dotnet-version: '3.1'
+          install-3: true
+          display-name: '.NET Core 3.1'
+          framework: 'netcoreapp3.1'
+          prefix: 'netcoreapp31'
+        - dotnet-version: '5.0'
+          install-3: false
+          display-name: '.NET 5.0'
+          framework: 'net5'
+          prefix: 'net5'
+    steps:
+    - uses: actions/checkout@v1
+    - name: Parse release version
+      run: python ./.github/scripts/get_release_version.py
+    - name: Setup .NET Core 3.1
+      uses: actions/setup-dotnet@v1
+      if: matrix.install-3
+      with:
+        dotnet-version: 3.1.x
+    - name: Setup .NET 5.0
```

Comment:
shouldn't this be checked against `install-3` being false?

Reply:
No. We need 5.0 regardless to build. The project here multi-target (both 3 and 5). 5 can build 3, but 3 can't build 5.